protocols/bizcoin/blockchain/messg.go

```diff
@@ -0,0 +1,132 @@
+package blockchain
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/dedis/cothority/lib/app"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/proof"
+	"github.com/dedis/cothority/protocols/bizcoin/blockchain/blkparser"
+	"github.com/dedis/crypto/abstract"
+)
+
+// Default port for the conode-setup - the stamping-request port
+// is at ```DefaultPort + 1```
+var DefaultPort int = 2000
+
+type MessageType int
+
+type SeqNo byte
+
+const (
+	Error MessageType = iota
+	TransactionAnnouncmentType
+	BlockReplyType
+	BlockRequestType
+	KeyBlockRequestType
+	BitCoSiClose
+	BitCoSiExit
+)
+
+type TransactionAnnouncment struct {
+	Val blkparser.Tx // Trasaction to be included in a block
+}
+
+// NOT: In order to decoe correctly the Proof, we need to the get the suite
+// somehow. We could just simply add it as a field and not (un)marhsal it
+// We'd just make sure that the suite is setup before unmarshaling.
+type BlockReply struct {
```

Comment:
So how is it done finally? 'We could' means: you didn't (at least not now). "We'd" means any of: we had, we should, or we would.

So where does the `SuiteStr` come from?